### ISSUE TITLE
FileIO integration

### ISSUE THREAD

**Issue by gdkrmr** [JuliaDataCubes/YAXArrays.jl#131]
Does it make sense to integrate this package with FileIO?

This way you can do
```
save("mycube.zarr", mycube)
```

- Can FileIO handle directories?
- Should the be done here or in Zarr.jl?
- I think this would also allow a much more general version of `@loadOrGenerate` https://github.com/JuliaIO/FileIO.jl/issues/359

**Comment by meggart** [comment]
I agree it would be great to do this. I once looked into this and it looks like this is difficult for directories since you can not read magic bytes from a directory. 

Regarding the loadorgenerate macro, I would be happy to remove it from YAXArrays, because the functionality is in no way specific to YAXArrays. I would be happy if this became a small package as you suggested in the FileIO issue, so we don't have to maintain it here.
